lib/logger.js

```diff
@@ -26,23 +44,65 @@ module.exports = (config) => {
       json: !notProd,
       timestamp: true,
       colorize: true,
+      format: winston.format.combine(
+        winston.format.timestamp(),
+        winston.format.json()
+      ),
       stringify: function stringify(obj) {
         return JSON.stringify(obj);
-      }
+      },
+      formatter: customFormatter
     })
   );
 
-  const transports = {
+  const levels = {
+    levels: {
+      error: 0,
+      warn: 1,
+      info: 2,
+      http: 3,
+      verbose: 4,
+      debug: 5,
+      silly: 6
+    }
+  };
+
+  const loggerConfig = {
+    levels: levels.levels,
+    level: config.logLevel,
     transports: loggingTransports,
     exceptionHandlers: exceptionTransports,
     exitOnError: true
   };
 
   if (notProd) {
-    delete transports.exceptionHandlers;
+    delete loggerConfig.exceptionHandlers;
   }
 
-  const logger = new winston.Logger(transports);
+  const logger = winston.createLogger(loggerConfig);
+
+  if (config.env === 'test') {
+    logger.silent = true;
+  }
+
+  const writeLogType = logLevel => {
+    return function logWrapper() {
+      const args = Array.from(arguments);
+      logger[logLevel](util.format(...args));
+    };
+  };
+
+  const allLogLevels = {
+    error: writeLogType('error'),
+    warn: writeLogType('warn'),
+    info: writeLogType('info'),
+    http: writeLogType('http'),
+    verbose: writeLogType('verbose'),
+    debug: writeLogType('debug'),
+    silly: writeLogType('silly')
+  };
 
-  return logger;
+  return Object.assign({}, allLogLevels, {
+    log: (level, ...args) => allLogLevels[level](...args)
```

Comment:
@AlexFranklinHODDaT so Marvel created their own dependency called Churchill which hasn't been updated in 4 years. At the moment it sits in the middle of 'hof' as a dependency that orchestrates the logging. We need to look at it closer. But I've updated winston to the latest version and the use of `logger.log` is for version 2.x.x not 3.x.x. To get round this I've added this to hook into what it did before, (i.e. 1st arg is logLevel and the rest of args are the message). Need to create a card to either remove or migrate code from winston. Also maybe to ask Ben why they set it up in the 1st place. It might have been to make debugging HOF easier.

Reply:
Ah, it might have been more about getting it setup so the logger is accessible on the request object too when hof is brought into a project. Will make this conversation done and leave these comments here for future reference.